builder

```diff
@@ -342,6 +348,32 @@ sub compile_packs {
     return $compiled;
 }
 
+sub fetch_portage_from_git {
+
+  my ($url, $srv, $branch ) = @_;
+
+  $url = substr($url, 0, length($url)-1) unless substr($url, -1) cmp "/";
+  my $uri = URI->new($url);
+  my $base = ( $uri->path_segments )[-1];
+
+  say "==== Fetching $srv Portage for branch $branch ====";
+
+  if ( $srv and $srv eq "gitlab" ) {
+    $url .= "/-/archive/$branch/$base-$branch.tar.gz";
+  } else {
+    $url .= "/archive/$branch.tar.gz";
+  }
+
+  _system( "wget -O /tmp/portage.tar.gz $url" );
+  _system( "mkdir /tmp/portage ; tar xzf /tmp/portage.tar.gz -C /tmp/portage" );
+
+  # Avoid to remove /usr/portage for use case where I have already
+  # files under /usr/portage/distfiles or /usr/portage/packages to use.
+  _system( "cp -arf /tmp/portage/$base-$branch/* /usr/portage/" );
```

Comment:
We can't just copy, that would break the portage tree present in the container:
If there is already one there and ebuilds are getting dropped in the version that you fetch, they will still be there ( and e.g. in case of thin manifests off, it would lead to broken manifests pretty fast)

Reply:
ok, I will fix this.